src/main/java/io/leonis/mx/Mx.java

```diff
@@ -0,0 +1,185 @@
+      return new Multiplex4.WithValue<>(this.preComp.apply(this.value), value -> value,
+          value -> multiplex.getFirstMux().apply(multiplex.getPreComp().apply(value)),
+          value -> multiplex.getSecondMux().apply(multiplex.getPreComp().apply(value)),
+          value -> multiplex.getThirdMux().apply(multiplex.getPreComp().apply(value)),
+          value -> multiplex.getFourthMux().apply(multiplex.getPreComp().apply(value)));
+    }
+
+    public <K0, M0, M1, M2, M3, M4> Multiplex5.WithValue<I1, I1, M0, M1, M2, M3, M4> expand(
+        final Multiplex5<I1, K0, M0, M1, M2, M3, M4> multiplex
+    ) throws Exception {
+      return new Multiplex5.WithValue<>(this.preComp.apply(this.value), value -> value,
+          value -> multiplex.getFirstMux().apply(multiplex.getPreComp().apply(value)),
+          value -> multiplex.getSecondMux().apply(multiplex.getPreComp().apply(value)),
+          value -> multiplex.getThirdMux().apply(multiplex.getPreComp().apply(value)),
+          value -> multiplex.getFourthMux().apply(multiplex.getPreComp().apply(value)),
+          value -> multiplex.getFifthMux().apply(multiplex.getPreComp().apply(value)));
+    }
+
+    public <K0, M0, M1, M2, M3, M4, M5> Multiplex6.WithValue<I1, I1, M0, M1, M2, M3, M4, M5> expand(
+        final Multiplex6<I1, K0, M0, M1, M2, M3, M4, M5> multiplex
+    ) throws Exception {
+      return new Multiplex6.WithValue<>(this.preComp.apply(this.value), value -> value,
+          value -> multiplex.getFirstMux().apply(multiplex.getPreComp().apply(value)),
+          value -> multiplex.getSecondMux().apply(multiplex.getPreComp().apply(value)),
+          value -> multiplex.getThirdMux().apply(multiplex.getPreComp().apply(value)),
+          value -> multiplex.getFourthMux().apply(multiplex.getPreComp().apply(value)),
+          value -> multiplex.getFifthMux().apply(multiplex.getPreComp().apply(value)),
+          value -> multiplex.getSixthMux().apply(multiplex.getPreComp().apply(value)));
+    }
+
+    public <K0, M0, M1, M2, M3, M4, M5, M6> Multiplex7.WithValue<I1, I1, M0, M1, M2, M3, M4, M5, M6> expand(
+        final Multiplex7<I1, K0, M0, M1, M2, M3, M4, M5, M6> multiplex
+    ) throws Exception {
+      return new Multiplex7.WithValue<>(this.preComp.apply(this.value), value -> value,
+          value -> multiplex.getFirstMux().apply(multiplex.getPreComp().apply(value)),
+          value -> multiplex.getSecondMux().apply(multiplex.getPreComp().apply(value)),
+          value -> multiplex.getThirdMux().apply(multiplex.getPreComp().apply(value)),
+          value -> multiplex.getFourthMux().apply(multiplex.getPreComp().apply(value)),
+          value -> multiplex.getFifthMux().apply(multiplex.getPreComp().apply(value)),
+          value -> multiplex.getSixthMux().apply(multiplex.getPreComp().apply(value)),
+          value -> multiplex.getSeventhMux().apply(multiplex.getPreComp().apply(value)));
+    }
+  }
+
+  @Value
+  final class MultiplexWithoutValue<I0, I1> {
```

Comment:
Does it need to be `@Value` ? You don't use the getters, or the equality, right? Could this be an explicitly final class with an `AllArgsConstructor`?

Reply:
the getters are used